### PR TITLE
M2-1987-2129 Gyroscope and touch

### PR DESCRIPTION
**Pull request by wildkiomi**

### CODE REVIEW

File: src/modules/Builder/features/PerformanceTasks/Flanker/Flanker.tsx [hunk 1]
```diff
@@ -8,6 +7,7 @@ import { PerformanceTaskHeader } from '../PerformanceTaskHeader';
 import { NameDescription } from '../NameDescription';
 import { GeneralSettings } from './GeneralSettings';
 import { RoundSettings, RoundUiType } from './RoundSettings';
+import { PerformanceTaskBody } from '../PerformanceTasks.styles';
```
Comment (kdanilenka32) [inline src/modules/Builder/features/PerformanceTasks/Flanker/Flanker.tsx:10]: pls use StyledPerf..

**Review by alex-i-e — APPROVED** [review]

**Review by kdanilenka32 — APPROVED** [review]

File: src/modules/Builder/features/PerformanceTasks/PerformanceTasks.styles.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { Box, styled } from '@mui/material';
+
+export const StyledPerformanceTaskBody = styled(Box)`
+  width: 83.2rem;
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/PerformanceTasks/PerformanceTasks.styles.ts:4]: for what do you need this strikt width?

File: src/modules/Builder/features/Activities/Activities.tsx [hunk 3]
```diff
@@ -183,7 +196,8 @@ export const Activities = () => {
                     const activityKey = getActivityKey(activity);
                     const isPerformanceTask = activity?.isPerformanceTask || false;
                     const activityName = activity.name;
-                    const isEditVisible = !isPerformanceTask || !!activity.isFlankerItem;
+                    const isEditVisible =
+                      !isPerformanceTask || activity.type === PerformanceTasks.Flanker;
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/Activities/Activities.tsx:200]: don't you need visible edit for touch and gyroscope?

File: src/modules/Builder/features/Activities/Activities.tsx [hunk 4]
```diff
@@ -197,8 +193,20 @@ export const Activities = () => {
                     const isPerformanceTask = activity?.isPerformanceTask || false;
                     const activityName = activity.name;
                     const isEditVisible =
-                      !isPerformanceTask || activity.type === PerformanceTasks.Flanker;
+                      !isPerformanceTask ||
+                      Object.values(EditablePerformanceTasks).includes(
+                        (activity.type || '') as any,
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/Activities/Activities.tsx:198]: don't use any type

File: src/modules/Builder/features/Activities/Activities.tsx [hunk 5]
```diff
@@ -197,8 +193,20 @@ export const Activities = () => {
...
+                      );
                     const hasError = errors[`activities[${index}]`];
+                    console.log(
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/Activities/Activities.tsx:201]: remove console.log

File: src/modules/Builder/pages/BuilderApplet/BuilderApplet.schema.ts [hunk 6]
```diff
@@ -121,16 +151,25 @@ export const ItemSchema = () =>
...
-              ? schema.required(getIsRequiredValidateMessage('correctAnswer'))
-              : schema,
-          ),
+      config: yup.object({}).when('responseType', (responseType, schema) => {
+        if (
+          responseType === ItemResponseType.Touch ||
+          responseType === ItemResponseType.Gyroscope
+        ) {
+          return schema.shape(GyroscopeAndTouchConfigSchema());
+        } else {
```
Comment (IhorNesterenko99) [inline src/modules/Builder/pages/BuilderApplet/BuilderApplet.schema.ts:160]: you don't need else if you use return

File: src/modules/Builder/pages/BuilderApplet/BuilderApplet.utils.tsx [hunk 7]
```diff
@@ -135,18 +136,36 @@ export const getNewPerformanceTask = ({
...
     [PerformanceTasks.Flanker]: defaultFlankerProps,
-    [PerformanceTasks.Gyroscope]: defaultGyroscopeAndTouchProps,
-    [PerformanceTasks.Touch]: defaultGyroscopeAndTouchProps,
+    [PerformanceTasks.Gyroscope]: {
+      responseType: ItemResponseType.Gyroscope,
+      ...defaultGyroscopeAndTouchProps,
+    },
+    [PerformanceTasks.Touch]: {
+      responseType: ItemResponseType.Touch,
+      defaultGyroscopeAndTouchProps,
```
Comment (IhorNesterenko99) [inline src/modules/Builder/pages/BuilderApplet/BuilderApplet.utils.tsx:145]: maybe also spread operator is needed?

Comment (IhorNesterenko99) [inline src/modules/Builder/pages/BuilderApplet/BuilderApplet.utils.tsx:145]: ...defaultGyroscopeAndTouchProps

File: src/modules/Builder/pages/BuilderApplet/BuilderApplet.utils.tsx [hunk 8]
```diff
@@ -135,18 +136,36 @@ export const getNewPerformanceTask = ({
...
+    },
   };
 
   const defaultPropsByType = type
-    ? propsByTypeObj[type as unknown as EditablePerformanceTasks] || {}
-    : {};
+    ? propsByTypeObj[type as unknown as EditablePerformanceTasks] || { responseType: '' }
+    : { responseType: '' };
+
+  const { responseType, ...config } = defaultPropsByType;
```
Comment (IhorNesterenko99) [inline src/modules/Builder/pages/BuilderApplet/BuilderApplet.utils.tsx:153]: I think responseType can be removed from defaultPropsByType

File: src/modules/Builder/pages/BuilderApplet/BuilderApplet.utils.tsx [hunk 9]
```diff
@@ -135,18 +136,36 @@ export const getNewPerformanceTask = ({
...
     name,
     description,
-    ...defaultPropsByType,
+    items: [
+      {
+        id: undefined,
+        key: uuidv4(),
+        name: `${responseType}`,
+        responseType,
+        question: '',
```
Comment (IhorNesterenko99) [inline src/modules/Builder/pages/BuilderApplet/BuilderApplet.utils.tsx:164]: I think question and responseValues can be removed from here, they are not mandatory

File: src/modules/Builder/features/PerformanceTasks/Instruction/InstructionContent/InstructionContent.tsx [hunk 10]
```diff
@@ -4,18 +4,18 @@ import { StyledBodyLarge, theme } from 'shared/styles';
 import { useCurrentActivity } from 'modules/Builder/hooks';
 import { EditorController, EditorUiType } from 'shared/components/FormComponents';
 
-import { OverviewInstructionProps } from '../OverviewInstruction.types';
+import { InstructionContentType } from './InstructionContent.types';
 
-export const InstructionContent = ({ description }: OverviewInstructionProps) => {
+export const InstructionContent = ({ description, name }: InstructionContentType) => {
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/PerformanceTasks/Instruction/InstructionContent/InstructionContent.tsx:9]:
```suggestion
export const InstructionContent = ({ description, name }: Omit<InstructionProps, 'title'>) => {
```

File: src/modules/Builder/features/PerformanceTasks/Instruction/InstructionContent/InstructionContent.types.ts
Comment (IhorNesterenko99) [inline src/modules/Builder/features/PerformanceTasks/Instruction/InstructionContent/InstructionContent.types.ts:1]: remove pls

File: src/modules/Builder/pages/BuilderApplet/BuilderApplet.schema.ts [hunk 11]
```diff
@@ -72,6 +78,30 @@ export const ResponseValuesNumberSelectionSchema = () => ({
...
+      .min(MIN_LENGTH_OF_TEST, <string>t('integerError'))
+      .max(MAX_LENGTH_OF_TEST, <string>t('integerError')),
+    lambdaSlope: yup
+      .number()
+      .min(MIN_SLOPE, <string>t('integerError'))
+      .max(MAX_SLOPE, <string>t('integerError')),
+  }),
+  practice: yup.object({
+    instructions: yup.string(),
+  }),
```
Comment (IhorNesterenko99) [inline src/modules/Builder/pages/BuilderApplet/BuilderApplet.schema.ts:99]:
```suggestion
  }).nullable(),
```
?

File: src/modules/Builder/pages/BuilderApplet/BuilderApplet.schema.ts [hunk 12]
```diff
@@ -72,6 +78,30 @@ export const ResponseValuesNumberSelectionSchema = () => ({
...
+  test: yup.object({
+    instructions: yup.string(),
+  }),
```
Comment (IhorNesterenko99) [inline src/modules/Builder/pages/BuilderApplet/BuilderApplet.schema.ts:102]: pls check if this field also can be nullable

File: src/modules/Dashboard/features/RespondentData/consts.ts
Comment (IhorNesterenko99) [inline src/modules/Dashboard/features/RespondentData/consts.ts:1]: rename pls to RespondentsData.consts.ts

**Pull request merged**